src/lddc.cpp

```diff
@@ -59,16 +65,33 @@ Lddc::Lddc(int format, int multi_topic, int data_src, int output_type,
   global_imu_pub_ = nullptr;
   cur_node_ = nullptr;
   bag_ = nullptr;
+
+  if (enable_dust_filter_)

```

Comment:
One more small comment:
You don't really need `enable_dust_filter_`. You can use the existence of `dust_filter_` instead. Since if `dust_filter_` is valid, `enable_...` must also be true. I think it makes the code a bit easier simpler, as checking 2 conditions later on.

You could also replace the `optional<vector>` by just check whether the vector is empty or not, but this is more of a style/verbosity choice